### PR TITLE
Check for local untracked files when deploying

### PR DESCRIPTION
**Pull request by LoryPack**
Way to fix issue #167.

### CODE REVIEW

**Comment by alshedivat** [comment]
Awesome, thank you!

File: bin/deploy [hunk 1]
```diff
@@ -62,6 +62,16 @@ if ! git diff-index --quiet HEAD --; then
     [[ "$0" = "$BASH_SOURCE" ]] && exit 1 || return 1
 fi
 
+# Check if there are any untracked files
+u="$(git ls-files --exclude-standard --others)"
+if ! test -z "$u"; then
```
Comment (alshedivat) [inline bin/deploy:67]:
```suggestion
if ! test -z "$(git ls-files --exclude-standard --others)"; then
```

**Review by alshedivat — APPROVED** [review]
Looks great, thanks for the fix! Please make a tiny stylistic change and I'll be happy to merge.

File: bin/deploy (hunk 1 above)
Comment (LoryPack) [reply bin/deploy:67]: Hi, thanks so much for that. There is actually a reason why I kept that separate (see here https://stackoverflow.com/a/2659808/13516418). Basically if you do not split the line there is a small chance that errors from git ls-files are translated as “no untracked files” reports, so that the deploy script may behave in weird ways which may be tricky for the user to understand. 

But if you think that is not an issue in practice I am happy to modify as you suggested!

Comment (alshedivat) [reply bin/deploy:67]: I see, thanks for clarifying.

It looks like separating the command into 2 lines makes the script treat the `git ls-files` error as if there were no untracked files. I would prefer the script to abort in such edge cases instead of proceeding silently.

Comment (LoryPack) [reply bin/deploy:67]: That makes sense. The above stackoverflow question seems to provide a way to address that: 

```
git ls-files --others --error-unmatch --exclude-standard . >/dev/null 2>&1; ec=$?
if test "$ec" = 0; then
    echo some untracked files
elif test "$ec" = 1; then
    echo no untracked files
else
    echo error from ls-files
fi
```

What do you think of that solution?

Comment (alshedivat) [reply bin/deploy:67]: In case when `git ls-files --others --error-unmatch --exclude-standard` returns an error code (which is an unlikely edge case), I believe the script will abort because of `set -e` on line 43 and won't get to the if-then-else part (to test this, we would need to come with a situation that results in `git ls-files` erroring out).

Btw, using `set -e` is not the best way to handle exceptions in bash (e.g., see #131). I plan to move away from `set -e` to proper exception handling. For now, I would prefer your original solution compressed to 1 line as I suggested above.

Comment (LoryPack) [reply bin/deploy:67]: Ok, thanks for explaining. I'll commit your previous suggestion then.
